test/shared/src/main/scala/zio/test/mock/Expectation.scala

```diff
@@ -16,9 +16,215 @@
 
 package zio.test.mock
 
+import scala.language.implicitConversions
+
+import com.github.ghik.silencer.silent
+import zio.{ IO, Managed, Ref, UIO, ZIO }
 import zio.test.Assertion
+import zio.test.mock.Expectation.{ AnyCall, Call, Empty, FlatMap, Next, State }
+import zio.test.mock.ReturnExpectation.{ Fail, Succeed }
+import zio.test.mock.MockException.UnmetExpectationsException
+
+/**
+ * A `Expectation[-M, +E, +A]` is an immutable data structure that represents
+ * expectations on module `M` capabilities.
+ *
+ * This structure is a monad, becouse we need the sequential composability and
+ * in Scala we get the the convinient for-comprehension syntax for free.
```

Comment:
"the" is repeated". "becouse" should be "because". "convinient" to "convenient".